library/core/src/ptr/mod.rs

```diff
@@ -2164,13 +2164,36 @@ pub fn addr_eq<T: ?Sized, U: ?Sized>(p: *const T, q: *const U) -> bool {
 
 /// Compares the *addresses* of the two function pointers for equality.
 ///
-/// Function pointers comparisons can have surprising results since
-/// they are never guaranteed to be unique and could vary between different
-/// code generation units. Furthermore, different functions could have the
-/// same address after being merged together.
+/// This is the same as `f == g`, but using this function makes clear that the potentially
+/// surprising semantics of function pointer comparison are involved.
+/// There are very few guarantees about how functions are compiled and they have no intrinsic
+/// “identity”; in particular, this comparison:
+///
+/// * May return `true` unexpectedly, in cases where functions are equivalent.
+///   For example, the following program is likely (but not guaranteed) to print `(true, true)`
+///   when compiled with optimization:
+///
+///   ```
```

Comment:
nit:
```suggestion
///   ```no_run
```

Reply:
I intentionally wrote this code with no assertions so that it will not panic. Would you prefer I make it an `assert!()` that isn't run?

Reply:
I though `no_run` added a small icon, like done for `compile_fail`, but that's apparently not the case.

As for `assert!` idk, doesn't feel necessary.